### ISSUE TITLE
Exception when I deserialize arrays in object within an class

### ISSUE THREAD

**Issue by lopper** [evermeer/EVReflection#141]
I get an exception when I attempt to deserialize an array of objects within a class. Here is my test example below. Please any advice would be appreciated.

```
public class ClassB: EVObject {
    public var name:String;
}

public class ClassA: EVObject {
    public var items:Array<ClassB> = []
}
```

When I run the code below and I get an exception deserializing the items property in ClassA.

```
func test()
{
        let dic = NSMutableDictionary()
        dic["name"] = "wow";

         let mainDic = NSMutableDictionary()
         mainDic["items"] = [dic];

        let item:ClassA = ClassA(dictionary: mainDic);
}
```

The exception I get traces to 

```
        if theValue is Array<Any> {
            return ((theValue as! Array<Any>) as AnyObject, valueType, false)
        }
```

in EVReflection.swift

**Comment by lopper** [comment]
I found a solution to the issue. I had to mark ClassB with **@objc(ClassB)**

In addition, I removed the cast to Array<Any> from the code where the exception is thrown. Here is the updated lines of code

```
if theValue is Array<Any> {
            return ((theValue as AnyObject, valueType, false)
}
```

**Comment by evermeer** [comment]
Strange that you get this error with the code above. What EVReflection version are you using?

I added your issue as a unit test and it just works:
https://github.com/evermeer/EVReflection/blob/master/EVReflection/EVReflectionTests/EVReflectionIssue141.swift#L26

I only had to change the ClassB name String to an optional because you did not initialize it.

**Comment by lopper** [comment]
I'm using version 3.4.1 of EVReflection.

I'm testing on OS X EL Capitan 10.11.6 with XCode 8.0 (8A218a)

**Comment by evermeer** [comment]
3.4.1 is indeed the latest version. I do use XCode 8.1, but that should not be the issue. 

If you just download it and run the tests, does the EVReflectionIssue141 test succeed?
https://github.com/evermeer/EVReflection/archive/master.zip

And if it succeeds, then whats different from your code?

**Comment by lopper** [comment]
Thanks for taking the time to look into the issue. 

I downloaded your code and attempted to run EVReflectionIssue141. However, I encountered a compilation issue. The code will not run on XCode 8.0. The code needs Swift 3.0.1 and XCode 8.0 only supports Swift 3.0.

I, then, upgrade my version of XCode to 8.1. I am an now able to run your test case and it runs successfully. In addition, the issue I was having with my own code seems to have disappeared. I assume that the issue is resolved in the Swift 3.0.1 update.

Thanks again.

**Comment by rishi420** [comment]
@evermeer sir if that case of swift 3.0.1 dependency wouldn't it be nice to have minimum required language to Swift 3.0.1 written somewhere e.g README. I have Xcode 8.0 faced same issue. Now downloading Xcode 8.1. BTW branch 'swift3' is working in Xcode 8.0. Thanks for this great repo. 👍 

Update: I confirm Xcode 8 can not parse array. It crashes. Xcode 8.1 works successfully.
